lib/widget/paragraph.cpp

```diff
@@ -0,0 +1,125 @@
+/*
+	This file is part of Warzone 2100.
+	Copyright (C) 2020  Warzone 2100 Project
+
+	Warzone 2100 is free software; you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation; either version 2 of the License, or
+	(at your option) any later version.
+
+	Warzone 2100 is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with Warzone 2100; if not, write to the Free Software
+	Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA
+*/
+
+#include "lib/framework/frame.h"
+#include "widget.h"
+#include "widgint.h"
+#include "paragraph.h"
+#include "form.h"
+#include "tip.h"
+
+#include <algorithm>
+
+class ParagraphLine: public WIDGET
+{
+public:
+	ParagraphLine(W_INIT *init): WIDGET(init, WIDG_UNSPECIFIED_TYPE) {}
+
+	Paragraph *getParagraph()
+	{
+		return (Paragraph *)parent();
+	}
+
+	void display(int xOffset, int yOffset)
+	{
+		text.render(xOffset + x(), yOffset + y() - text.aboveBase(), getParagraph()->fontColour);
+	}
+
+	void setText(std::string newText)
```

Comment:
`const std::string&`?

Reply:
arent we supposed to use the "wzstring" @past-due ?